plugins/logging/file

```diff
@@ -230,7 +230,9 @@ sub maybe_reopen {
         unless ($self->open_log($new_output, $transaction)) {
             return undef;
         }
-        $transaction->notes('file-reopened-this-session', 1);
+        if (UNIVERSAL::can($transaction,'isa')) {
```

Comment:
just a few lines above:
```
my $new_output = $self->log_output($transaction);
```
and in log_output:
```
 $output =~ s/%i/($transaction->notes('logging-session-id') || 'parent')/ge;
```
I suppose we want to do the same magic to avoid calling undef->notes() in log_output()?